lib/services/internalEngine/index.js

```diff
@@ -488,12 +488,13 @@ class InternalEngine {
    *
    * @param {string} type - data collection
    * @param {object} mapping
+   * @param {string} index - data index (default to this.index)
    * @return {Promise}
    */
-  updateMapping (type, mapping) {
+  updateMapping (type, mapping, index = this.index) {
     const request = {
       type,
-      index: this.index,
+      index: index,
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use shorthand for property "index". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3498)